config.js

```diff
@@ -1,83 +1,81 @@
+      db = require('./db'),
+      logger = require('./lib/logger'),
+      defaults = require('./defaults'),
+      translations = {},
+      SETTINGS_PATH = '_design/medic/_rewrite/app_settings/medic';
 
-function loadTranslations() {
-    var options = {
-        startkey: [ 'translations', false ],
-        endkey: [ 'translations', true ],
-        include_docs: true
-    };
-    db.medic.view('medic-client', 'doc_by_type', options, function(err, result) {
-        if (err) {
-            logger.error('Error loading translations - starting up anyway', err);
-            return;
-        }
-        result.rows.forEach(function(row) {
-            translations[row.doc.code] = row.doc.values;
-        });
-    });
-}
+let config = require('./defaults');
 
-function initFeed() {
-    /*
-     * Use since=now on ddoc listener so we don't replay an old change.
-     */
-    var feed = new follow.Feed({ db: process.env.COUCH_URL, since: 'now' });
-
-    /*
-     * Hack until figure out a better way to reload app settings in all
-     * the calling contexts.
-     */
-    feed.on('change', function(change) {
-        if (change.id === '_design/medic') {
-            logger.debug('change event on doc %s seq %s', change.id, change.seq);
-            logger.info('reload triggered, exiting...');
-            process.exit(0);
-        } else if (change.id.indexOf('messages-') === 0) {
-            logger.info('Detected translations change - reloading');
-            loadTranslations();
-        }
+const loadTranslations = function() {
+  const options = {
+    startkey: [ 'translations', false ],
+    endkey: [ 'translations', true ],
+    include_docs: true
+  };
+  db.medic.view('medic-client', 'doc_by_type', options, function(err, result) {
+    if (err) {
+      logger.error('Error loading translations - starting up anyway', err);
+      return;
+    }
+    result.rows.forEach(function(row) {
+      translations[row.doc.code] = row.doc.values;
     });
+  });
+};
 
-    feed.follow();
-}
+const initFeed = function() {
+  // Use since=now on ddoc listener so we don't replay an old change.
+  const feed = new follow.Feed({ db: process.env.COUCH_URL, since: 'now' });
+  feed.on('change', function(change) {
+    if (change.id === '_design/medic') {
+      logger.info('Reloading configuration');
+      initConfig(function(err) {
+        if (err) {
+          console.error('Error loading configuration. Exiting...');
+          process.exit(0);
+        }
+        require('./transitions').loadTransitions();
+      });
+    } else if (change.id.indexOf('messages-') === 0) {
+      logger.info('Detected translations change - reloading');
+      loadTranslations();
+    }
+  });
+  feed.follow();
+};
 
-function initConfig(data, callback) {
-    var settings = data.settings;
-    // merge defaults with app settings
-    _.extend(config, settings);
+const initConfig = function(callback) {
+  db.medic.get(SETTINGS_PATH, function(err, data) {
```

Comment:
I'm going to stop mentioning these, but IMO you should go through and change them all :-)